src/components/Dashboard/FilingHistoryList/HeaderActions.vue

```diff
@@ -125,18 +125,32 @@ export default class HeaderActions extends Vue {
     // see also TodoList.vue:doResumeFiling()
     switch (filing?.name) {
       case FilingTypes.ALTERATION:
-      case FilingTypes.INCORPORATION_APPLICATION:
       case FilingTypes.CHANGE_OF_REGISTRATION:
       case FilingTypes.CORRECTION:
+      case FilingTypes.INCORPORATION_APPLICATION:
       case FilingTypes.REGISTRATION:
         // correction via Edit UI
         this.mutateCurrentFiling(filing)
         this.mutateFileCorrectionDialog(true)
         break
 
-      case FilingTypes.ANNUAL_REPORT:
       case FilingTypes.CHANGE_OF_ADDRESS:
       case FilingTypes.CHANGE_OF_DIRECTORS:
+        if (this.isBenBcCccUlc) {
+          // correction via Edit UI if current type is BC, CC, ULC, or BEN
+          this.mutateCurrentFiling(filing)
```

Comment:
I know there's a bit of code duplication here but I couldn't think of a cleaner way to achieve this.

Reply:
This doesn't bother me.